lib/util/url.ts

```diff
@@ -106,3 +108,26 @@ export function parseLinkHeader(
   }
   return _parseLinkHeader(linkHeader);
 }
+
+export function urlContainsSubPath(url: string | undefined): boolean {
+  const parsedUrl = parseUrl(url);
+  if (is.nullOrUndefined(parsedUrl)) {
+    return false;
+  }
+  // url has subpath(nested path)
+  // if there is more than than one slash in pathname with a string after it.
+  const lastSlashPos = parsedUrl.pathname.lastIndexOf('/');
+  const afterLastSlash = parsedUrl.pathname.substring(lastSlashPos);
+  return (
+    lastSlashPos > 0 &&
+    !is.emptyString(afterLastSlash) &&
+    afterLastSlash !== '/'
+  );
```

Comment:
can this be done using regex check?

Reply:
Which do you think they'll prefer ?

Reply:
not sure, but imo regex is less code and more understandable, plus less `index out of bound` issues